From d4be8d51f6785228f53acf7303d6ff4e4b8fabd6 Mon Sep 17 00:00:00 2001
From: Yangbo Lu <yangbo.lu@nxp.com>
Date: Mon, 19 Jun 2017 18:46:04 +0800
Subject: [PATCH 179/767] dpaa_eth: move global variable 'clock' into
 ptp_priv_s struct

This patch is to move global variable 'clock' for DPAA PTP
clock pointer into ptp_priv_s struct.

Signed-off-by: Yangbo Lu <yangbo.lu@nxp.com>
[Xulin: Original patch taken from NXP LSDK-18.09.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h | 1 +
 drivers/net/ethernet/freescale/sdk_dpaa/dpaa_ptp.c | 5 +++--
 2 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h
index eeb80c7ee540..84d0afed1db0 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_eth.h
@@ -669,6 +669,7 @@ static inline void _dpa_assign_wq(struct dpa_fq *fq)
 struct ptp_priv_s {
 	struct device_node *node;
 	struct platform_device *of_dev;
+	struct ptp_clock *clock;
 	struct mac_device *mac_dev;
 };
 extern struct ptp_priv_s ptp_priv;
diff --git a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_ptp.c b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_ptp.c
index 7f12ba0c9a6f..f54a3d679425 100644
--- a/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_ptp.c
+++ b/drivers/net/ethernet/freescale/sdk_dpaa/dpaa_ptp.c
@@ -27,8 +27,6 @@
 #include "dpaa_eth.h"
 #include "mac.h"
 
-struct ptp_clock *clock;
-
 static struct mac_device *mac_dev;
 static u32 freqCompensation;
 
@@ -256,6 +254,7 @@ static int __init __cold dpa_ptp_load(void)
 {
 	struct device *ptp_dev;
 	struct timespec64 now;
+	struct ptp_clock *clock = ptp_priv.clock;
 	int dpa_phc_index;
 	int err;
 
@@ -283,6 +282,8 @@ module_init(dpa_ptp_load);
 
 static void __exit __cold dpa_ptp_unload(void)
 {
+	struct ptp_clock *clock = ptp_priv.clock;
+
 	if (mac_dev->fm_rtc_disable_interrupt)
 		mac_dev->fm_rtc_disable_interrupt(mac_dev->fm_dev, 0xffffffff);
 	ptp_clock_unregister(clock);
-- 
2.17.0

